src/snyk/cli/services/cliService.ts

```diff
@@ -66,4 +85,26 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
 
     return args;
   }
+
+  public async isChecksumCorrect(cliPath: string): Promise<boolean> {
+    if (this.verifiedChecksumCorrect !== undefined) {
+      return this.verifiedChecksumCorrect;
+    }
+
+    const downloadedChecksum = this.extensionContext.getGlobalStateValue<string>(MEMENTO_CLI_CHECKSUM);
+    if (!downloadedChecksum) {
+      throw new Error('Checksum not found in the global storage.');
+    }
+
+    const checksum = await Checksum.getChecksumOf(cliPath, downloadedChecksum);
+    this.verifiedChecksumCorrect = checksum.verify();
+
+    return this.verifiedChecksumCorrect;
```

Comment:
let's try to have one `return` line in this function

Reply:
what's the reasoning behind? I can wrap the whole logic in a big if-else but it wouldn't be very readable if more nesting will be introduced in future.